### ISSUE TITLE
Require exactly n non-option arguments

### ISSUE THREAD

**Issue by tremby** [bcoe/yargs#169]
Is there a way to require exactly n non-option arguments? `.require(2)` right now means "require at least 2 non-option arguments", but I want an error if 3 are provided.

Obviously I can do this by checking the length of `argv._`, but it'd be nice if it were built in.

**Comment by bcoe** [comment]
@tremby as far as I know, there's not currently a way to get this functionality. What would make sense to me, would be for `strict()` to enforce this behavior. So:

```
var argv = require('yargs')
  .demand(2)
  .strict()
  .argv
```

Would raise an exception unless exactly two arguments are provided, along with continuing to raise exceptions if unrecognized arguments are provided.

**Comment by bcoe** [comment]
@tremby what do you think of this approach?

https://github.com/bcoe/yargs/pull/182

If you like it, mind giving the code a spin and letting me know if you find any bugs.

**Comment by tremby** [comment]
Seems reasonable. Nicely backwards-compatible. Realistically I won't have time to test this any time soon. But you can go ahead and close this if it all checks out for you.

**Comment by bcoe** [comment]
@tremby I've merged #182, and pushed the changes to the `next` tag on npm. If you do get a sec, run:

`npm install yargs@next`

either way, the changes will be available soon in `yargs@latest`.
